### ISSUE TITLE
Hovering tooltip works but navigation fails with a variable scoped with a typedef

### ISSUE THREAD

**Issue by LeThiHyVong** [microsoft/vscode-cpptools#1968]
Issue Type: <b>Bug</b>

For some external definition of member variables and enumeration, hovering tooltip displayed correctly but navigation (Go To Definition, Peek Definition, Go to Declaration, Peek Declaration) all failed. 
However, they worked well on macro.

Captures:
![tooltip](https://user-images.githubusercontent.com/19317679/39794293-d0a55ae2-5373-11e8-92e8-71afec5380b1.png)
![no definition](https://user-images.githubusercontent.com/19317679/39794295-d1ccfe84-5373-11e8-99a4-97065bf8d512.png)

Sample source code:

[source.zip](https://github.com/Microsoft/vscode-cpptools/files/1986291/source.zip)

Extension version: 0.17.0
VS Code version: Code 1.23.0 (7c7da59c2333a1306c41e6e7b68d7f0caa7b3d45, 2018-05-03T16:44:10.813Z)
OS version: Windows_NT ia32 6.1.7601

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-4590 CPU @ 3.30GHz (4 x 3292)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>video_decode: enabled<br>video_encode: enabled<br>vpx_decode: unavailable_software<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|3.42GB (0.23GB free)|
|Process Argv|C:\Program Files\Microsoft VS Code\Code.exe|
|Screen Reader|no|
|VM|0%|

</details>

**Comment by sean-mcmanus** [comment]
Yeah, this looks like a bug. Go to Definition works with s.npos (where s is a string).

**Comment by sean-mcmanus** [comment]
The typedef for string is breaking this, i.e. go to definition works on std::basic_string<char>::npos, but fails with foo_t in:

```
class foo
{
public:
    static int mFoo;
};

typedef foo foo_t;

int main() {
    foo_t::mFoo; // Go to def fails
    foo::mFoo; // Go to def works
    return 0;
}
```

UPDATE: Looks like this is currently "by design" -- we need to bring in more components from VS 2017 deal with this. No ETA yet.

**Comment by sean-mcmanus** [comment]
The Go to Definition failing with std::string::npos is fixed with 0.19.0, but the modified example I came up with still has a bug, which I'm tracking with https://github.com/Microsoft/vscode-cpptools/issues/2574 .

UPDATE: It got fixed with https://github.com/Microsoft/vscode-cpptools/releases/tag/0.19.1-insiders .
